common/src/main/java/org/apache/rocketmq/common/Configuration.java

```diff
@@ -219,6 +224,35 @@ public void persist() {
         }
     }
 
+    public void persistBrokerConf(Properties from) {
+        BufferedReader reader = null;
+        try {
+            readWriteLock.readLock().lockInterruptibly();
+            String fileName = this.getStorePath();
+            File file = new File(fileName);
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            Properties properties = new Properties();
+            reader = new BufferedReader(new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8));
+            properties.load(reader);
+            merge(from, properties);
+            final String str = MixAll.properties2String(properties);
+            MixAll.string2File(str, fileName);
+        } catch (Exception e) {
+            log.error("persist brokerConf error", e);
```

Comment:
Maybe when an exception is thrown, we also need to close the reader?

Reply:
Thanks for your review,i will fix it